File: Youtube_python/channel_bd.py
from channel import Channel
from video import Video

class ChannelBd(Channel):
    def __init__(self):
        super().__init__()
        self.channel_list = []
    
    def add_channel(self, channel):
        self.channel_list.append(channel)
        print(f"Chaîne créée avec l'ID: {channel.channel_id}")
    
    def connection_channel(self):
        name = input("Entrez le nom de votre chaine :")
        for channel in self.channel_list:
            if channel.host_name == name:
                print(f"CONNEXION REUSSIE. Bienvenu sur {channel.name_channel}")
                return
        print("Impossible de se connecter a la chaine. veuillez entrer le bon nom ou creer une nouvelle chaine")
    
    def search_channel(self, nom):
        for channel in self.channel_list:
            if channel.name_channel == nom:
                print("\n========================================")
                print(f"\t \tID Chaîne: {channel.channel_id}")
                print(f"\t \tNom de la chaine: {channel.name_channel}")
                print(f"\t \tPropietaire de la chaine: {channel.host_name}")
                print(f"\t \tFollowers: {channel.followers}")
                print(f"\t \tNombre de Video: {channel.videos_number}")
                print(f"\t \tDate de creation: {channel.date_creation}")
                print("========================================\n")
                self.display_list_video()
                return
        print(f"Chaîne introuvable: {nom}")
    
    def follow_channel(self, titre):
        for channel in self.channel_list:
            if channel.name_channel == titre:
                channel.followers += 1
                print(f"La chaine '{titre}' a été suivi ! Nouveau nombre de followers : {channel.followers}")
                return
        print(f"Chaine introuvable : {titre}")
    
    def unfollow_channel(self, titre):
        for channel in self.channel_list:
            if channel.name_channel == titre:
                channel.followers -= 1
                print(f"La chaine '{titre}' a arrete d'etre suivi par un follower ! Nouveau nombre de followers : {channel.followers}")
                return
        print(f"Chaine introuvable : {titre}")
    
    def posted_video(self, video):
        self.nbre_video.append(video)
        self.videos_number += 1
        print(f"Vidéo postée avec succès! ID: {video.video_id}")
    
    def delete_video(self, video_id):
        for video in self.nbre_video:
            if video.video_id == video_id:
                self.nbre_video.remove(video)
                self.videos_number -= 1
                print(f"Vidéo {video_id} supprimée avec succès.")
                return
        print(f"Vidéo introuvable: {video_id}")
    
    def display_channel(self):
        if not self.channel_list:
            print("Aucune chaîne disponible.")
            return
        
        for channel in self.channel_list:
            print("\n========================================")
            print(f"\t \tID Chaîne: {channel.channel_id}")
            print(f"\t \tNom de la chaine: {channel.name_channel}")
            print(f"\t \tPropietaire de la chaine: {channel.host_name}")
            print(f"\t \tFollowers: {channel.followers}")
            print(f"\t \tNombre de Video: {channel.videos_number}")
            print(f"\t \tDate de creation: {channel.date_creation}")
            print("========================================\n")

File: Youtube_c++/ChannelBd.cpp
#include "ChannelBd.h"
#include <iostream>

ChannelBd::ChannelBd() : Channel() {}

void ChannelBd::addChannel(const Channel& c) {
    channelList.push_back(c);
    std::cout << "Chaîne créée avec l'ID: " << c.channelId << "\n";
}

void ChannelBd::connectionChannel() {
    std::string name;
    std::cout << "Entrez le nom de votre chaine :\n";
    std::getline(std::cin, name);
    
    for (const auto& c : channelList) {
        if (c.hostName == name) {
            std::cout << "CONNEXION REUSSIE. Bienvenu sur " << c.nameChannel << "\n";
            return;
        }
    }
    std::cout << "Impossible de se connecter a la chaine. veuillez entrer le bon nom ou creer une nouvelle chaine\n";
}

void ChannelBd::searchChannel(const std::string& nom) {
    for (const auto& c : channelList) {
        if (c.nameChannel == nom) {
            std::cout << "\n========================================\n";
            std::cout << "\t \tID Chaîne: " << c.channelId << "\n";
            std::cout << "\t \tNom de la chaine: " << c.nameChannel << "\n";
            std::cout << "\t \tPropietaire de la chaine: " << c.hostName << "\n";
            std::cout << "\t \tFollowers: " << c.followers << "\n";
            std::cout << "\t \tNombre de Video: " << c.videosNumber << "\n";
            std::cout << "\t \tDate de creation: " << c.dateCreation << "\n";
            std::cout << "========================================\n\n";
            displayListVideo();
            return;
        }
    }
    std::cout << "Chaîne introuvable: " << nom << "\n";
}

void ChannelBd::followChannel(const std::string& titre) {
    for (auto& c : channelList) {
        if (c.nameChannel == titre) {
            c.followers++;
            std::cout << "La chaine '" << titre << "' a été suivi ! Nouveau nombre de followers : " << c.followers << "\n";
            return;
        }
    }
    std::cout << "Chaine introuvable : " << titre << "\n";
}

void ChannelBd::unfollowChannel(const std::string& titre) {
    for (auto& c : channelList) {
        if (c.nameChannel == titre) {
            c.followers--;
            std::cout << "La chaine '" << titre << "' a arrete d'etre suivi par un follower ! Nouveau nombre de followers : " << c.followers << "\n";
            return;
        }
    }
    std::cout << "Chaine introuvable : " << titre << "\n";
}

void ChannelBd::postedVideo(const Video& v) {
    nbreVideo.push_back(v);
    videosNumber++;
    std::cout << "Vidéo postée avec succès! ID: " << v.videoId << "\n";
}

void ChannelBd::deleteVideo(const std::string& id) {
    for (auto it = nbreVideo.begin(); it != nbreVideo.end(); ++it) {
        if (it->videoId == id) {
            nbreVideo.erase(it);
            videosNumber--;
            std::cout << "Vidéo " << id << " supprimée avec succès.\n";
            return;
        }
    }
    std::cout << "Vidéo introuvable: " << id << "\n";
}

void ChannelBd::displayChannel() const {
    if (channelList.empty()) {
        std::cout << "Aucune chaîne disponible.\n";
        return;
    }
    
    for (const auto& c : channelList) {
        std::cout << "\n========================================\n";
        std::cout << "\t \tID Chaîne: " << c.channelId << "\n";
        std::cout << "\t \tNom de la chaine: " << c.nameChannel << "\n";
        std::cout << "\t \tPropietaire de la chaine: " << c.hostName << "\n";
        std::cout << "\t \tFollowers: " << c.followers << "\n";
        std::cout << "\t \tNombre de Video: " << c.videosNumber << "\n";
        std::cout << "\t \tDate de creation: " << c.dateCreation << "\n";
        std::cout << "========================================\n\n";
    }
}
